mls-rs/Cargo.toml

```diff
@@ -55,12 +55,11 @@ zeroize = { version = "1", default-features = false, features = ["alloc", "zeroi
 mls-rs-codec = { version = "0.5.0", path = "../mls-rs-codec", default-features = false}
 thiserror = { version = "1.0.40", optional = true }
 itertools = { version = "0.12.0", default-features = false, features = ["use_alloc"]}
-enum-iterator = "1.1.3"
 cfg-if = "1"
 debug_tree = { version = "0.4.0", optional = true }
-spin = { version = "0.9.8", default-features = false, features = ["mutex", "spin_mutex", "portable_atomic"]}
-portable-atomic = { version = "1.5.1", default-features = false, features = ["critical-section"]}
-portable-atomic-util = { version = "0.1.2", default-features = false, features = ["alloc"]}
+spin = { version = "0.9.8", default-features = false, features = ["mutex", "spin_mutex", "portable_atomic"], optional = true}
+portable-atomic = { version = "1.5.1", default-features = false, features = ["critical-section"], optional = true}
+portable-atomic-util = { version = "0.1.2", default-features = false, features = ["alloc"], optional = true}
```

Comment:
Right... and from what I understand and [can read](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification), you're doing things the recommended way by having a `std` feature instead of a `no_std` one.

I wonder what the usual work-around is in this case. The portable-atomic and portable-atomic-util crates are the ones I'm trying to avoid since I don't have them available in AOSP where I work.

Reply:
Let me drop this part for now: I can patch things locally as needed and I think you're correct that there is no current good way of doing what I want.

Reply:
I think we can't rely on features (at least as we've defined them) to do that. Other options are:
1. Adding these dependencies based on the platform -- this may be quite tedious and incomplete.
2. Adding another cfg like `mls_build_async` to enable/disable these dependencies.

I can check (2) with Tom and get back to you. This can be done in a separate PR if that helps.